### PR TITLE
docs: add icon resource page

### PR DESCRIPTION
**Pull request by hamza14khan**
- Addition of icon resource page.
- Showcase of icons.

Screenshot 2024-10-29 at 17 59 46

### CODE REVIEW

**Review by BogdanDanilescu — APPROVED** [review]
LGTM

**Review by ramonmulia — APPROVED** [review]
LGTM

File: apps/docs/src/components/document/resources/icon-grid-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  'search',
+  'menu',
+  'home',
+  'logout',
+  'download',
+];
+
+export function IconGridLayout() {
+  return (
+    <div className="grid grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-6 border-t border-l">
```
Comment (pregno) [inline apps/docs/src/components/document/resources/icon-grid-layout.tsx:24]: max 4 columns

Comment (pregno) [inline apps/docs/src/components/document/resources/icon-grid-layout.tsx:24]: add a `gap` here, so you can get rid of margin between cards

**Review by pregno — CHANGES_REQUESTED** [review]

**Review by pregno — APPROVED** [review]

**Pull request merged**